CorsixTH/Lua/hospital.lua

```diff
@@ -256,6 +256,13 @@ function Hospital:Hospital(world, avail_rooms, name)
   end
 end
 
+--! Checks if a room has been discovered
+--!param room_id (string) The name of the room
+--!return (boolean) true if found
+function Hospital:isRoomDiscovered(room_id)
+  return room_id and self.room_discovery[room_id].is_discovered
```

Comment:
Why do we protect against nil / false room_id here (but not otherwise invalid room_ids)?

Reply:
I'm not sure if you're saying we should remove the room_id check or whether we should check room_id is a string?

Reply:
I wasn't saying to do anything, I was asking why their was a nil check / what sort of bad input we expect.